lib/kinesis_client.rb

```diff
@@ -96,7 +90,21 @@ def push_batch(batch)
   end
 
   def push_records
-    @batch.each_slice(@batch_size) {|slice| push_batch slice}
-    @batch = []
+    if @records.length > 0 && @records.length < @batch_size
+      push_batch(@records)
+    else
+      @records.each_slice(@batch_size) { |slice| push_batch(slice) }
+    end
+    @records = []
+  end
+
+  # where should I put this?
```

Comment:
I would guess this would be called by the receiver? We'd also want to add some tests for this.